### ISSUE TITLE
Make a wiki

### ISSUE THREAD

**Issue by littlewoo** [littlewoo/tgds#44]
We need the wiki to cover the following things:
- setting up for development (pretty much just git and maven... everything else is optional)
- explanation of the architecture
- documentation for the api

**Comment by littlewoo** [comment]
This issue also covers creating diagrams explaining architecture, and making sure that all the code has proper javadoc.

**Comment by littlewoo** [comment]
It is of course, entirely possible that in trying to document the architecture, I'll discover just how awful it is. If that happens, this issue is also going to involve some refactoring.

**Comment by bexedmondson** [comment]
Would it be worth making new issues for any problems you find?

**Comment by littlewoo** [comment]
Yes. I am discovering at this point that it may be quite a large amount of work! :P
